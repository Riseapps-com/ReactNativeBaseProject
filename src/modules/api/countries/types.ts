export type NativeNameItem = {
  common: string;
  official: string;
};

export type NativeName = {
  [key: string]: NativeNameItem;
};

export type Name = NativeNameItem & {
  nativeName: NativeName;
};

export type Currency = {
  name: string;
  symbol: string;
};

export type Currencies = {
  [key: string]: Currency;
};

export type Idd = {
  root: string;
  suffixes: string[];
};

export type Languages = {
  [key: string]: string;
};

export type Translation = {
  official: string;
  common: string;
};

export type Translations = {
  [key: string]: Translation;
};

export type Demonym = {
  f: string;
  m: string;
};

export type Demonyms = {
  [key: string]: Demonym;
};

export type Maps = {
  googleMaps: string;
  openStreetMaps: string;
};

export type Gini = {
  [key: string]: number;
};

export type Flags = {
  svg: string;
  png: string;
};

export type Country = {
  name: Name;
  tld: string[];
  cca2: string;
  ccn3: string;
  cca3: string;
  cioc: string;
  independent: boolean;
  status: string;
  unMember: boolean;
  currencies?: Currencies;
  idd: Idd;
  capital?: string[];
  altSpellings: string[];
  region: string;
  subregion: string;
  languages?: Languages;
  translations?: Translations;
  latlng: number[];
  landlocked: boolean;
  borders: string[];
  area: number;
  demonyms: Demonyms;
  flag: string;
  maps: Maps;
  population: number;
  gini: Gini;
  flags: Flags;
};

export type Region = 'africa' | 'americas' | 'asia' | 'europe' | 'oceania';
