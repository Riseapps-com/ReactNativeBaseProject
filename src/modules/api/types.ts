export type Currency = {
  code: string;
  name: string;
  symbol: string;
};

export type Language = {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  iso639_1: string;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  iso639_2: string;
  name: string;
  nativeName: string;
};

export type Translations = {
  de: string;
  es: string;
  fr: string;
  ja: string;
  it: string;
  br: string;
  pt: string;
  nl: string;
  hr: string;
  fa: string;
};

export type RegionalBloc = {
  acronym: string;
  name: string;
  otherAcronyms: string[];
  otherNames: string[];
};

export type Country = {
  id: string;
  name: string;
  topLevelDomain: string[];
  alpha2Code: string;
  alpha3Code: string;
  callingCodes: string[];
  capital: string;
  altSpellings: string[];
  region: string;
  subregion: string;
  population: number;
  latlng: number[];
  demonym: string;
  area: number;
  gini: number;
  timezones: string[];
  borders: string[];
  nativeName: string;
  numericCode: string;
  currencies: Currency[];
  languages: Language[];
  translations: Translations;
  flag: string;
  regionalBlocs: RegionalBloc[];
  cioc: string;
};

export type Region = 'africa' | 'americas' | 'asia' | 'europe' | 'oceania';
