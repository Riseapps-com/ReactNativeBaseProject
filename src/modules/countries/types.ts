export type LocalCountry = {
  id: string;
  name: string;
  capital: string;
  region: string;
  subregion: string;
  timezones: string[];
  currencies: string[];
  alpha2Code: string;
};
