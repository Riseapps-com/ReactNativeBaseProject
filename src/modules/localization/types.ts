export type TranslationGetters = {
  [key: string]: () => Record<string, any>;
};
