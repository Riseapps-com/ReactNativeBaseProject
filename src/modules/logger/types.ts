import type * as SentryTypes from '@sentry/types';

export type SentryMessageDetails = SentryTypes.Context | SentryTypes.Contexts;
