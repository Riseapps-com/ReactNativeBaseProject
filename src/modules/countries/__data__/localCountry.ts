import { LocalCountry } from '../types';

const localCountry: Omit<LocalCountry, 'id'> = {
  cca2: 'UA',
  name: 'Ukraine',
  capital: 'Kyiv',
  region: 'Europe',
  subregion: 'Eastern Europe',
  population: '44,134,693',
  flagLink: 'https://flagcdn.com/w320/ua.png',
  currencies: 'Ukrainian hryvnia',
  languages: 'Ukrainian',
};

export { localCountry };
