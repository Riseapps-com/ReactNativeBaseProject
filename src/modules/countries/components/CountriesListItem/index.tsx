import React, { useCallback } from 'react';
import { TouchableOpacity, View } from 'react-native';

import { testIDs } from '~config';
import { LocalCountry } from '~modules/state';
import { FastImage, resizeMode, Text } from '~modules/ui';
import { useTheme } from '~theme';

import { countriesUtils } from '../../services';
import themedStyles from './styles';

export type CountriesListItemProps = {
  country: LocalCountry;
  index: number;
  onItemPress?: (index: number) => void;
};

const CountriesListItem: React.FC<CountriesListItemProps> = props => {
  const [styles] = useTheme(themedStyles);
  const { onItemPress } = props;

  const handleItemPress = useCallback(
    () => onItemPress && onItemPress(props.index),
    [onItemPress, props.index]
  );

  return (
    <TouchableOpacity
      testID={testIDs.countries.country}
      onPress={handleItemPress}
      style={styles.container}
    >
      <FastImage
        style={styles.icon}
        resizeMode={resizeMode.contain}
        source={{ uri: countriesUtils.getCountryFlag(props.country.alpha2Code, 'h60') }}
      />

      <View style={styles.centerContainer}>
        <Text numberOfLines={1} fontStyle={'bold'}>
          {props.country.name}
        </Text>
        <Text numberOfLines={1} style={styles.subtitle} fontStyle={'bold'}>
          {props.country.capital || '-'}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default CountriesListItem;
