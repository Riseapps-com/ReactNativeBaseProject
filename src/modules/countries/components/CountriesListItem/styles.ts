import { sizes, styleSheetFactory } from '~theme';

export default styleSheetFactory(theme => ({
  container: {
    flexDirection: 'row',
    paddingHorizontal: sizes.SPACING_M,
    paddingVertical: sizes.SPACING_S,
  },

  centerContainer: {
    flex: 1,
    marginStart: sizes.SPACING_M,
  },

  subtitle: {
    color: theme.disabled,
  },

  icon: {
    width: sizes.SPACING_XL,
  },
}));
