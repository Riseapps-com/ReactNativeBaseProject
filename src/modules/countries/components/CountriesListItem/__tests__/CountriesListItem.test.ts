import { fireEvent } from '@testing-library/react-native';

import { renderComponent } from '~modules/tests';

import { localCountry } from '../../../__data__';
import { LocalCountry } from '../../../types';
import CountriesListItem, { CountriesListItemProps } from '../index';

const renderCountriesListItem = (props: CountriesListItemProps) => renderComponent(CountriesListItem, props);

describe('countries', () => {
  describe('<CountriesListItem />', () => {
    it('renders <CountriesListItem />', () => {
      const countriesListItem = renderCountriesListItem({
        country: localCountry,
        index: 0,
      });

      expect(countriesListItem.getByText(localCountry.name)).toBeTruthy();
      expect(countriesListItem).toMatchSnapshot();
    });

    it('renders placeholder if capital is empty', () => {
      const country: LocalCountry = {
        ...localCountry,
        capital: '',
      };
      const countriesListItem = renderCountriesListItem({
        country,
        index: 0,
      });

      expect(countriesListItem.getByText('-')).toBeTruthy();
    });

    it('calls on item press', () => {
      const onItemPress = jest.fn();
      const index = 4;
      const countriesListItem = renderCountriesListItem({
        country: localCountry,
        index,
        onItemPress,
      });

      fireEvent.press(countriesListItem.getByA11yLabel('Countries list item'));
      jest.runAllTimers();

      expect(onItemPress).toBeCalledTimes(1);
      expect(onItemPress).toBeCalledWith(index);
    });
  });
});
