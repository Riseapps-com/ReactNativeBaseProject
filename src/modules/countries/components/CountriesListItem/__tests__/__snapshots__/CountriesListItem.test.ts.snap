// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countries <CountriesListItem /> renders <CountriesListItem /> 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="Countries list item"
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://flagcdn.com/w320/ua.png",
      }
    }
    style={
      Object {
        "width": 32,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginStart": 16,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "#4A4E4D",
            },
            Object {
              "fontFamily": "SF Pro Text Bold",
            },
            Object {
              "fontSize": 14,
            },
            undefined,
          ],
        ]
      }
    >
      Ukraine
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "#4A4E4D",
            },
            Object {
              "fontFamily": "SF Pro Text Bold",
            },
            Object {
              "fontSize": 14,
            },
            Object {
              "color": "rgba(74, 78, 77, 0.5)",
            },
          ],
        ]
      }
    >
      Kyiv
    </Text>
  </View>
</View>
`;
